docs/source/notes/autograd.rst

```diff
@@ -86,6 +36,137 @@ flow statements, that can change the overall shape and size of the graph at
 every iteration. You don't have to encode all possible paths before you
 launch the training - what you run is what you differentiate.
 
+.. _locally-disable-grad-doc:
+
+Locally disabling gradient computation
+--------------------------------------
+
+There are several ways to locally disable gradients: setting
+``requires_grad``, enabling no-grad mode, and enabling inference mode. Below, we
+describe the different mechanisms and how they are used. We also describe a
+similar mechanism, ``.eval()`` (evaluation mode), that is often mixed up with
+the three.
+
+Default Mode (Grad Mode)
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+All operations that have at least one input with ``requires_grad=True`` will recorded
+in the backward graph by default.
+This allows us to compute gradients when running ``output.backward()``
+as described above in :ref:`how-autograd-encodes-history`.
+
+This default "mode" is the mode we are implicitly in when no other modes like
+no-grad and inference mode are enabled.
+Unlike the other modes, grad mode is also the only mode where ``requires_grad`` is not overridden.
+In fact, the primary purpose of no-grad mode is to override ``requires_grad``.
+
+To be contrasted with "no-grad mode" the default mode is sometimes called "grad mode".
+
+Setting the ``requires_grad`` flag of leaf tensors
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:attr:`requires_grad` is a flag that allows for fine-grained exclusion of subgraphs
+from gradient computation.
+Every tensor has this flag, but setting this flag really only makes sense for leaf
+tensors (tensors that do not have a ``grad_fn``, e.g., a module's parameters).
+Non-leaf tensors (tensors that do have ``grad_fn``) are actually defined to always
+have ``requires_grad=True``.
+
+Setting ``requires_grad`` can take effect in both the forward and backward passes:
+During the forward pass, an operation is only recorded in the backward graph if at least one
+of its input tensors require grad. The output of this operation is then defined to
+have ``require_grad=True`` as well since it is a non-leaf tensor.
+During ``.backward()``, only leaf tensors with ``requires_grad=True`` will have
+gradients accumulated into their ``.grad`` fields.
+
+Since it is a common pattern, ``requires_grad`` flags can also be applied to modules
+via :meth:`nn.Module.requires_grad_()`.
+When applied to a module, ``.requires_grad_()`` takes effect on all
+of the module's parameters (which set ``requires_grad=True`` by default).
+
+Setting ``requires_grad`` should be the main way you should control which parts of the model
+are part of the gradient computation. For example, setting ``requires_grad``
+is used if you need to freeze part of your pretrained model when finetuning.
+To freeze a module, simply apply ``module.requires_grad_(False)``.
+As this flag is propagated to the module's parameters, computations in the forward pass
```

Comment:
To change to:
```
As this flag is propagated to the module's parameters, none of the inputs would require grad. As a result, computations in the forward pass will not be recorded. In the backward pass, gradients are not computed with respect to the module's parameters, and updates would not be made to them on the optimizer step, as desired.
```
Seems like a long and unwieldy series of steps though. Maybe bullet points here would be better?